source/SharedCrtResourceManager.h

```diff
@@ -42,7 +43,8 @@ namespace Aws
                 std::shared_ptr<Crt::Mqtt::MqttConnection> connection;
                 aws_allocator *allocator{nullptr};
                 aws_mem_trace_level memTraceLevel{AWS_MEMTRACE_NONE};
-                std::vector<Feature *> *features;
+                Util::FeatureRegistry *features;
```

Comment:
We want to avoid naked pointers except for C runtime objects such as `aws_allocator` and so forth. Make this a shared pointer to `FeatureRegistry`.

```
std::shared_ptr<Util::FeatureRegistry> features;
```